### PR TITLE
Back port of Task Thin Execution List and List By Name

### LINKED ISSUE #5995: Port #5994 to 2.11.x

### PR DESCRIPTION
**Pull request by corneil**
Updates api docs for tasks/thinexecutions/name and all HATEOAS links. Updates related tests for links. Improved link handling and tests in TaskTemplate.
Original PR #5994 
Issues #5991, #5973

Fixes #5995

### CODE REVIEW

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java [hunk 1]
```diff
@@ -234,6 +235,9 @@ public void index() throws Exception {
 
 						fieldWithPath("_links.tasks/thinexecutions.href").description("Link to the tasks/thinexecutions"),
 
+						fieldWithPath("_links.tasks/thinexecutions/name.href").description("Link to the tasks/thinexecutions/name"),
+						fieldWithPath("_links.tasks/thinexecutions/name.templated").type(JsonFieldType.BOOLEAN).optional().description("Link to the tasks/thinexecutions/name is templated"),
```
Comment (onobc) [inline spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java:239]: The optional makes sense but I noticed it was not in the source being ported ([here](https://github.com/spring-cloud/spring-cloud-dataflow/pull/5994/files#diff-9158f2aa24bd9b317120f16404df0a58043b47e3479e147777f71dba6e36cba8R225)).

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java [hunk 2]
```diff
@@ -124,6 +124,7 @@ public void index() throws Exception {
 				linkWithRel("tasks/platforms").description("Provides platform accounts for launching tasks.  The results can be filtered to show the platforms that support scheduling by adding a request parameter of 'schedulesEnabled=true"),
 				linkWithRel("tasks/logs").description("Retrieve the task application log"),
 				linkWithRel("tasks/thinexecutions").description("Returns thin Task executions"),
+				linkWithRel("tasks/thinexecutions/name").description("Returns all thin Task executions for a given Task name"),
```
Comment (onobc) [inline spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java:127]: I noticed in the source being ported there was a removal of `linkWithRel("tasks/thinexecutions")` ([here](https://github.com/spring-cloud/spring-cloud-dataflow/pull/5994/files#diff-9158f2aa24bd9b317120f16404df0a58043b47e3479e147777f71dba6e36cba8L125)). Is this needed here as well?

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java [hunk 3]
```diff
@@ -234,6 +235,9 @@ public void index() throws Exception {
 
 						fieldWithPath("_links.tasks/thinexecutions.href").description("Link to the tasks/thinexecutions"),
 
+						fieldWithPath("_links.tasks/thinexecutions/name.href").description("Link to the tasks/thinexecutions/name"),
+						fieldWithPath("_links.tasks/thinexecutions/name.templated").type(JsonFieldType.BOOLEAN).optional().description("Link to the tasks/thinexecutions/name is templated"),
+
 						fieldWithPath("_links.tasks/schedules.href").description("Link to the tasks/executions/schedules"),
```
Comment (onobc) [inline spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java:241]: I noticed the source has a removal for `fieldWithPath("_links.tasks/thinexecutions.href")` ([here](https://github.com/spring-cloud/spring-cloud-dataflow/pull/5994/files#diff-9158f2aa24bd9b317120f16404df0a58043b47e3479e147777f71dba6e36cba8L230)).

File: spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc [hunk 4]
```diff
@@ -1862,6 +1862,8 @@ The following topics provide more details:
 * <<api-guide-resources-task-executions-stopping>>
 * <<api-guide-resources-task-executions-list>>
 * <<api-guide-resources-task-executions-list-by-name>>
+* <<api-guide-resources-task-thin-executions-list>>
```
Comment (onobc) [inline spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc:1865]: On the source port, there was some concerns about the docs not rendering properly ([here](https://github.com/spring-cloud/spring-cloud-dataflow/pull/5994/files#r1813314218)). Can you verify this has been fixed or is not an issue here as well?

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java [hunk 5]
```diff
@@ -158,6 +162,11 @@ public class TaskTemplate implements TaskOperations {
 			this.thinExecutionsLink = null;
 		}
 
+		if(resources.getLink(THIN_EXECUTIONS_BY_NAME_RELATION).isPresent()) {
```
Comment (onobc) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:165]: This can be simplified w/ optional such as:
```java
this.thinExecutionsByNameLink = resources.getLink(THIN_EXECUTIONS_BY_NAME_RELATION).orElse(null);
```
This is fine as well though and its a backport.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java [hunk 6]
```diff
@@ -283,14 +292,23 @@ public TaskExecutionResource.Page executionList() {
...
 		} else {
 			return restTemplate.getForObject(executionsLink.getHref(), TaskExecutionThinResource.Page.class);
 		}
 	}
 
+	@Override
+	public TaskExecutionThinResource.Page thinExecutionListByTaskName(String taskName) {
+		if(thinExecutionsByNameLink != null) {
+			return restTemplate.getForObject(thinExecutionsByNameLink.expand(taskName).getHref(), TaskExecutionThinResource.Page.class);
+		} else {
```
Comment (onobc) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:307]: No need to have else, the last statement can just return.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RootController.java [hunk 7]
```diff
@@ -159,6 +159,7 @@ public RootResource info() {
 			root.add(linkTo(methodOn(TasksInfoController.class).getInfo(null, null, null)).withRel("tasks/info/executions"));
 			root.add(linkTo(methodOn(TaskLogsController.class).getLog(null, null, null)).withRel("tasks/logs"));
 			root.add(linkTo(methodOn(TaskExecutionThinController.class).listTasks(null, null)).withRel("tasks/thinexecutions"));
+			root.add(linkTo(methodOn(TaskExecutionThinController.class).retrieveTasksByName(null, null, null)).withRel("tasks/thinexecutions/name"));
 			if (featuresProperties.isSchedulesEnabled()) {
```
Comment (onobc) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RootController.java:163]: Not related to this file, but rather `TaskExecutionController`... the port source has this added code https://github.com/spring-cloud/spring-cloud-dataflow/pull/5994/files#diff-2cc7b0286a71d935467884c16b01bf672821257fc6005c25df49141dbd8e7b7dR188-R191

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionControllerTests.java [hunk 8]
```diff
@@ -374,6 +374,17 @@ void getAllThinExecutions() throws Exception {
 			.andExpect(jsonPath("$._embedded.taskExecutionThinResourceList", hasSize(4)));
 	}
 
+
+	@Test
```
Comment (onobc) [inline spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionControllerTests.java:378]: Why does the source test differ https://github.com/spring-cloud/spring-cloud-dataflow/pull/5994/files#diff-a37ece622810b7f245fa769ce9ee2df687e2c8f9ab859daa8fb8bbe04f74f16fR344 ?

File: src/scripts/build-docs.sh [hunk 9]
```diff
@@ -5,6 +5,6 @@ if [ -z "$BASH_VERSION" ]; then
 fi
 SCDIR=$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")
 SCDIR=$(realpath $SCDIR)
-pushd "$SCDIR" > /dev/null || exit
+pushd "$SCDIR/../.." > /dev/null || exit
```
Comment (onobc) [inline src/scripts/build-docs.sh:8]: I don't see this in the port source? What is this doing?

**Review by onobc — CHANGES_REQUESTED** [review]
Thanks for the back port @corneil . I have some questions. I notice there is also quite a few files modified in the source that is not in this PR. Please describe the discrepancies.

Thanks

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java (hunk 2 above)
Comment (corneil) [reply spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java:127]: There was a duplicate

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java (hunk 3 above)
Comment (corneil) [reply spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java:241]: It was a duplication.

File: spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc (hunk 4 above)
Comment (corneil) [reply spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc:1865]: I checked the rendering on the docs.

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java (hunk 1 above)
Comment (onobc) [reply spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java:239]: Do you mind updating main after this w/ the same?

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java (hunk 6 above)
Comment (corneil) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:307]: It's an old habit to have an else.

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionControllerTests.java (hunk 8 above)
Comment (corneil) [reply spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionControllerTests.java:378]: I need to merge the two sets of functionality.

**Comment by corneil** [comment]
I have resolved comments. There are a few elements I'll port to main to get parity.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java [hunk 10]
```diff
@@ -300,8 +300,8 @@ public TaskExecutionController taskExecutionController(
 		}
 
 		@Bean
-		public TaskExecutionThinController taskExecutionThinController(AggregateTaskExplorer aggregateTaskExplorer, TaskJobService taskJobService) {
-			return new TaskExecutionThinController(aggregateTaskExplorer, taskJobService);
+		public TaskExecutionThinController taskExecutionThinController(AggregateTaskExplorer aggregateTaskExplorer, TaskDefinitionRepository taskDefinitionRepository, TaskJobService taskJobService) {
```
Comment (onobc) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java:303]: What is this change for?

Comment (onobc) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java:303]: nm I see below

File: src/scripts/build-docs.sh (hunk 9 above)
Comment (onobc) [reply src/scripts/build-docs.sh]: Still curious about this one.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java (hunk 10 above)
Comment (corneil) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java:303]: If the search is for a task where we have no definition of that name then return 404. Otherwise the result is an empty list.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java [hunk 11]
```diff
@@ -145,14 +146,43 @@ public class TaskTemplate implements TaskOperations {
...
+			this.executionsCurrentLink = null;
+		}
+		if (VersionUtils.isDataFlowServerVersionGreaterThanOrEqualToRequiredVersion(version, VALIDATION_RELATION_VERSION)) {
+			Assert.isTrue(resources.getLink(VALIDATION_REL).isPresent(), ()->VALIDATION_REL + " relation is required");
+			this.validationLink = resources.getLink(VALIDATION_REL).get();
+		} else {
+			this.validationLink = null;
+		}
 
-		this.validationLink = resources.getLink(VALIDATION_REL).get();
```
Comment (onobc) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java]: Why these drastic changes in this subsequent review commit? I'm just not sure what this is in response to?

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java [hunk 12]
```diff
@@ -145,14 +146,43 @@ public class TaskTemplate implements TaskOperations {
...
-		this.aboutLink = resources.getLink("about").get();
 
-		this.thinExecutionsLink = resources.getLink(THIN_EXECUTIONS_RELATION).orElse(null);
-		this.thinExecutionsByNameLink = resources.getLink(THIN_EXECUTIONS_BY_NAME_RELATION).orElse(null);
-		this.executionLaunchLink = resources.getLink(EXECUTION_LAUNCH_RELATION).orElse(null);
-		this.executionsCurrentLink = resources.getLink(EXECUTIONS_CURRENT_RELATION).orElse(null);
+		if(VersionUtils.isDataFlowServerVersionGreaterThanOrEqualToRequiredVersion(version, VALIDATION_THIN_TASK_VERSION) && resources.getLink(THIN_EXECUTIONS_RELATION).isPresent()) {
```
Comment (onobc) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:164]: I would consider moving this logic into a well-named helper method to avoid the repeat checks as its hard to read currently.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java [hunk 13]
```diff
@@ -141,22 +146,33 @@ public class TaskTemplate implements TaskOperations {
 
 		this.restTemplate = restTemplate;
 		this.dataFlowServerVersion = dataFlowServerVersion;
-
 		String version = VersionUtils.getThreePartVersion(dataFlowServerVersion);
+		if (VersionUtils.isDataFlowServerVersionGreaterThanOrEqualToRequiredVersion(version, EXECUTIONS_CURRENT_RELATION_VERSION)) {
```
Comment (onobc) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:150]: Is this related to thinexecutions? If not, I would remove it from this PR.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java [hunk 14]
```diff
@@ -165,12 +181,7 @@ public class TaskTemplate implements TaskOperations {
 			this.executionLaunchLink = null;
 		}
 
-		if (VersionUtils.isDataFlowServerVersionGreaterThanOrEqualToRequiredVersion(version, EXECUTIONS_CURRENT_RELATION_VERSION)) {
```
Comment (onobc) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:168]: Why did this move up? If its not required to be at the top, please keep it in place so as to avoid confusing during the review.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java [hunk 15]
```diff
@@ -283,18 +294,26 @@ public TaskExecutionResource.Page executionList() {
 	}
 
 	@Override
-	public PagedModel<TaskExecutionThinResource> thinExecutionList() {
+	public TaskExecutionThinResource.Page thinExecutionList() {
```
Comment (onobc) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:297]: Signature change? Seems breaking.

File: src/scripts/build-docs.sh [hunk 16]
```diff
@@ -4,7 +4,7 @@ if [ -z "$BASH_VERSION" ]; then
     exit 1
 fi
 SCDIR=$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")
-SCDIR=$(realpath $SCDIR)
-pushd "$SCDIR" > /dev/null || exit
+SCDIR=$(realpath "$SCDIR/../..")
```
Comment (onobc) [inline src/scripts/build-docs.sh:7]: I still don't understand what this has to do w/ thin executions

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java (hunk 5 above)
Comment (corneil) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java]: Done where appropriate

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java (hunk 1 above)
Comment (corneil) [reply spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java:239]: Done

File: src/scripts/build-docs.sh (hunk 9 above)
Comment (corneil) [reply src/scripts/build-docs.sh]: Fixed for improved readability.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java (hunk 12 above)
Comment (corneil) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:164]: Removed the `resources.getLink(THIN_EXECUTIONS_RELATION).isPresent()` from if

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java (hunk 14 above)
Comment (corneil) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:168]: I fixed the order to match the previous version.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java (hunk 15 above)
Comment (corneil) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:297]: The result is the same type.

File: src/scripts/build-docs.sh (hunk 16 above)
Comment (corneil) [reply src/scripts/build-docs.sh:7]: I'll move this to a separate PR

File: src/scripts/build-docs.sh (hunk 9 above)
Comment (corneil) [reply src/scripts/build-docs.sh]: Reverting the change and moving to separate PR

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java (hunk 14 above)
Comment (onobc) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:168]: I would prefer change history over matching order in a different branch. Please undo the change.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java [hunk 17]
```diff
@@ -66,16 +65,17 @@ public class TaskTemplate implements TaskOperations {
 
 	private static final String EXECUTIONS_CURRENT_RELATION_VERSION = "1.7.0";
 
-	private static final String VALIDATION_RELATION_VERSION = "1.7.0";
```
Comment (onobc) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:69]: Why is this deleted? Please add a comment.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java (hunk 15 above)
Comment (corneil) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:288]: https://github.com/spring-cloud/spring-cloud-dataflow/blob/464868b87ab9a322a6dc7a2c9584475bbcfab8dc/spring-cloud-dataflow-rest-resource/src/main/java/org/springframework/cloud/dataflow/rest/resource/TaskExecutionThinResource.java#L185

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java [hunk 18]
```diff
@@ -134,51 +136,51 @@ public class TaskTemplate implements TaskOperations {
...
-			this.validationLink = resources.getLink(VALIDATION_REL).get();
+		if (VersionUtils.isDataFlowServerVersionGreaterThanOrEqualToRequiredVersion(version, EXECUTIONS_CURRENT_RELATION_VERSION)) {
+			Assert.isTrue(resources.getLink(EXECUTIONS_CURRENT_RELATION).isPresent(), ()-> EXECUTIONS_CURRENT_RELATION + " relation is required");
+			this.executionsCurrentLink = resources.getLink(EXECUTIONS_CURRENT_RELATION).get();
 		} else {
-			this.validationLink = null;
+			this.executionsCurrentLink = null;
 		}
 
-		// TODO early 2.11.3-SNAPSHOT version didn't have this. Remove && resources.getLink(THIN_EXECUTIONS_RELATION).isPresent() when releasing 2.11.3
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:153]: We should have removed the comment and if `resources.getLink(THIN_EXECUTIONS_RELATION).isPresent()` before 2.11.4 but didn't.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java (hunk 17 above)
Comment (corneil) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:69]: I restored that.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java (hunk 15 above)
Comment (onobc) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:288]: Well if they are the same let's leave it as is since this is a simple backport on what will be a patch release. What is the value in changing it?

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java (hunk 17 above)
Comment (onobc) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:69]: I don't see it.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java [hunk 19]
```diff
@@ -134,51 +136,51 @@ public class TaskTemplate implements TaskOperations {
 				EXECUTION_RELATION_BY_NAME,
 				EXECUTIONS_INFO_RELATION,
 				PLATFORM_LIST_RELATION,
-				RETRIEVE_LOG
+				RETRIEVE_LOG,
+				VALIDATION_REL
```
Comment (onobc) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:140]: What is this now?

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java [hunk 20]
```diff
@@ -134,51 +136,51 @@ public class TaskTemplate implements TaskOperations {
...
 		} else {
-			this.validationLink = null;
+			this.executionsCurrentLink = null;
 		}
 
-		// TODO early 2.11.3-SNAPSHOT version didn't have this. Remove && resources.getLink(THIN_EXECUTIONS_RELATION).isPresent() when releasing 2.11.3
-		if(VersionUtils.isDataFlowServerVersionGreaterThanOrEqualToRequiredVersion(version, VALIDATION_THIN_TASK_VERSION) && resources.getLink(THIN_EXECUTIONS_RELATION).isPresent()) {
+		this.aboutLink = resources.getLink("about").get();
+
+		this.definitionsLink = resources.getLink(DEFINITIONS_RELATION).get();
```
Comment (onobc) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:157]: What is all this?

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java (hunk 14 above)
Comment (corneil) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:168]: Will a squash into on commit not fix this so that the diff is smaller because the order is that same as before the PR branch?

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java (hunk 19 above)
Comment (corneil) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:140]: This was rolled back in the version I want to push.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java (hunk 15 above)
Comment (corneil) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java:295]: This is a lot cleaner as a result `return restTemplate.getForObject(executionsLink.getHref(), TaskExecutionThinResource.Page.class);`
